src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,190 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        add(value, size);
     }
 
     @Override
     public void add(T value, int index) {
+        validateIndexExclusive(index);
+        Node<T> newNode = new Node<>(value, null, null);
+        if (isEmpty()) {
+            head = newNode;
+            tail = newNode;
+        } else if (index == 0) {
+            newNode.next = head;
+            head.prev = newNode;
+            head = newNode;
+        } else if (index == size) {
+            newNode.prev = tail;
+            tail.next = newNode;
+            tail = newNode;
```

Comment:
Why do you need so much code here if you have **insertBefore** method?
```suggestion
        if (index == size) {
            add(value);
```